_docker/drupal/drupal-filesystem/scripts/drupal_install_checker.rb

```diff
@@ -21,148 +21,107 @@ def initialize(drupal_site, process_executor = ProcessExecutor.new, opts)
-                           'entity_storage_migrate', 'key_value', 'multiversion', 'token', 'metatag',
-                           'metatag_google_plus', 'metatag_open_graph', 'metatag_twitter_cards',
-                           'metatag_verification', 'admin_toolbar', 'admin_toolbar_tools', 'simple_sitemap']
+    tables_in_db = process_executor.exec!('mysql', ["--host=#{@opts['database']['host']}",
+                                                    "--port=#{@opts['database']['port']}",
+                                                    "--user=#{@opts['database']['username']}",
+                                                    "--password=#{@opts['database']['password']}",
+                                                    '--execute=show tables', "#{@opts['database']['name']}"])
 
-    if @opts['environment'] == 'dev'
-      module_install_args.push(*%w(devel kint))
+    tables_in_db.split("\n").each do | existing_table |
+      required_tables.delete_if do | required_table |
+        required_table == existing_table.chomp
+      end
     end
-    process_executor.exec!('/var/www/drupal/vendor/bin/drupal', module_install_args)
+
+    raise StandardError, "Not all required database tables are present. The missing tables are '#{required_tables}'. Aborting boot." unless required_tables.empty?
+    puts "\tAll required database tables are present."
+
   end
 
-  #
-  # This sets the cron key to a known value for the Drupal instance so that we can invoke cron remotely
-  #
-  def set_cron_key
-    puts 'Setting the cron key...'
-    process_executor.exec!('/var/www/drupal/vendor/bin/drupal', %w(--root=web state:override system.cron_key rhd))
+  def check_all_required_drupal_configuration!
+    settings_exists!
+    rhd_settings_exists!
+    database_tables_exists!
   end
 
-  def install_drupal
-    puts 'Installing Drupal, please wait...'
-    process_executor.exec!('/var/www/drupal/vendor/bin/drush',
-                           ['--root=/var/www/drupal/web', 'si', 'standard', '--locale=en',
-                            "--db-url=mysql://#{@opts['database']['username']}:#{@opts['database']['password']}@#{@opts['database']['host']}:#{@opts['database']['port']}/#{@opts['database']['name']}",
-                            '--site-name=Red Hat Developers', '--site-mail=test@example.com',
-                            "--account-name=#{@opts['drupal']['admin']['name']}",
-                            "--account-pass=#{@opts['drupal']['admin']['password']}",
-                            "--account-mail=#{@opts['drupal']['admin']['mail']}",
-                            '--config-dir=/var/www/drupal/web/config/sync',
-                            '-y'])
+  def drush_clear_cache
+    process_executor.exec!('/var/www/drupal/vendor/bin/drush', %w(--root=/var/www/drupal/web cr))
   end
 
-  def update_db
-    puts 'Executing drush dbup'
+  def drush_update_db
+    puts 'Executing drush dbup...'
```

Comment:
this should actually be updb

Reply:
we could even drop it and manually do an updb when we know we need to.

Reply:
@LightGuard I will re-name to reflect that this is actually `updb` as you say.

In terms of removing the `updb` call entirely from the boot sequence of Drupal, can you remind me what was the motivation behind doing that?

We would need to be aware that doing that will cause us a headache with the PR pipeline in that people would have to make sure that they ran the `updb` on their PR Drupal instance each time their PR was built if they are pushing changes to Drupal or the modules we use. I suspect people would forget to do that, particularly in the case of a "retest this please".

cheers,

Rob

Reply:
@LightGuard OK I understand.

At the minute I don't really see us running the `updb` as a problem as it only takes ~10 seconds when there are no changes to apply. As long as it's not breaking anything (which I assume it isn't as we've been doing it for a while), then my preference would be to leave it as is simply because it means that at the minute nobody needs to remember to apply any changes. 

I think the biggest challenge of making it a manual action would be ensuring people have access and remember to apply it on their PR builds.  We would ideally want people to apply any module modifications before the export starts so that we can ensure a module update doesn't break parts of the site. We'd therefore be looking at some sort of manual step in the PR pipeline that says "Have you run updb if you need to?" . The build would be paused until it is confirmed `updb` has been applied. It's possible to do that with Jenkins pipeline builds, but it doesn't feel like a great addition to the PR workflow.

Perhaps another alternative is that we somehow provide a flag to indicate to the Drupal boot process that it should apply `updb`  this time around, but again this feels like something to consider when running `updb` is causing a problem?

cheers,

Rob

Reply:
Okay, that's fair. We can keep it in.